experiment/logviewer/gcsreader/local.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"time"
+)
+
+// These methods are only meant to be used for local machine benchmarking, so they may be poor optimized or violate good style
+const testFilePath = "c:\\temp\\kube-apiserver.log"
+
+func setupForLocal() {
+	downloadToFiles()
+}
+
+// We assume read from file is fast - according to 'testRead' it takes ~1s on uncompressed file
+func runBenchmarks() {
+	testDecompress()
+	testDownload()
+	testParse()
+	testFileRead()
+}
+
+func testDecompress() {
+	defer timeTrack(time.Now(), "decompress")
+	r := readFromLocalFile(testFilePath + ".gz")
+	r, err := decompress(r)
+	io.Copy(ioutil.Discard, r)
+	handle(err)
+}
+
+func testDownload() {
+	defer timeTrack(time.Now(), "download")
+
+	r, err := download(testLogPath)
+	handle(err)
+	io.Copy(ioutil.Discard, r)
+}
+
+func testParse() {
+	defer timeTrack(time.Now(), "parse")
+	regex, _ := regexp.Compile(testTargetSubstring)
+	reader := bufio.NewReader(readFromLocalFile(testFilePath))
+	parsed, _ := processLines(reader, regex)
+	if len(parsed) == 0 {
+	}
+}
+
+func testFileRead() {
+	defer timeTrack(time.Now(), "read")
+	r := readFromLocalFile(testFilePath)
+	io.Copy(ioutil.Discard, r)
+}
+
+func downloadToFiles() {
+	reader, err := download(testLogPath)
+	handle(err)
```

Comment:
I moved this function to local.go file(which may be just safely deleted in future). I decided to keep it only for usage in this file, because it is playground and I'm not sure if it becomes better with inlining 10 usages of this function each producing 3 lines of code. However if you still don't like it, I can remove it even from playground. Same with timetracking - I kept 1 small function to use in playground and moved it there.